test/e2e/kubectl/kubectl.go

```diff
@@ -1320,46 +1319,47 @@ metadata:
 				return strings.Split(strings.TrimRight(out, "\n"), "\n")
 			}
 
-			ginkgo.By("Waiting for Redis master to start.")
-			waitForOrFailWithDebug(1)
-			forEachPod(func(pod v1.Pod) {
-				ginkgo.By("checking for a matching strings")
-				_, err := framework.LookForStringInLog(ns, pod.Name, containerName, "Ready to accept connections", framework.PodStartTimeout)
-				framework.ExpectNoError(err)
+			ginkgo.By("Waiting for log generator to start.")
+			if !e2epod.CheckPodsRunningReadyOrSucceeded(c, ns, []string{podName}, framework.PodStartTimeout) {
+				e2elog.Failf("Pod %s was not ready", podName)
+			}
 
-				ginkgo.By("limiting log lines")
-				out := framework.RunKubectlOrDie("log", pod.Name, containerName, nsFlag, "--tail=1")
-				gomega.Expect(len(out)).NotTo(gomega.BeZero())
-				framework.ExpectEqual(len(lines(out)), 1)
-
-				ginkgo.By("limiting log bytes")
-				out = framework.RunKubectlOrDie("log", pod.Name, containerName, nsFlag, "--limit-bytes=1")
-				framework.ExpectEqual(len(lines(out)), 1)
-				framework.ExpectEqual(len(out), 1)
-
-				ginkgo.By("exposing timestamps")
-				out = framework.RunKubectlOrDie("log", pod.Name, containerName, nsFlag, "--tail=1", "--timestamps")
-				l := lines(out)
-				framework.ExpectEqual(len(l), 1)
-				words := strings.Split(l[0], " ")
-				gomega.Expect(len(words)).To(gomega.BeNumerically(">", 1))
-				if _, err := time.Parse(time.RFC3339Nano, words[0]); err != nil {
-					if _, err := time.Parse(time.RFC3339, words[0]); err != nil {
-						e2elog.Failf("expected %q to be RFC3339 or RFC3339Nano", words[0])
-					}
+			ginkgo.By("checking for a matching strings")
+			_, err := framework.LookForStringInLog(ns, podName, "", "/api/v1/namespaces/kube-system", framework.PodStartTimeout)
```

Comment:
Can we keep the `containerName`? The test explicitly specifies the name; I think we should try preserving the behavior by default.

The same applies to the tests below.